lighthouse-treemap/app/src/main.js

```diff
@@ -9,6 +9,244 @@
+      if (!(e.target instanceof HTMLElement)) return;
+      const nodeEl = e.target.closest('.webtreemap-node');
+      if (!nodeEl) return;
+      this.updateColors();
+    });
+
+    window.addEventListener('mouseover', (e) => {
+      if (!(e.target instanceof HTMLElement)) return;
+      const nodeEl = e.target.closest('.webtreemap-node');
+      if (!nodeEl) return;
+      nodeEl.classList.add('webtreemap-node--hover');
+    });
+
+    window.addEventListener('mouseout', (e) => {
+      if (!(e.target instanceof HTMLElement)) return;
+      const nodeEl = e.target.closest('.webtreemap-node');
+      if (!nodeEl) return;
+      nodeEl.classList.remove('webtreemap-node--hover');
+    });
+  }
+
+  /**
+   * @param {string} name
+   */
+  findRootNode(name) {
+    for (const rootNodes of Object.values(this.rootNodesByGroup)) {
+      for (const rootNode of rootNodes) {
+        if (rootNode.name === name) return rootNode;
+      }
+    }
+  }
+
+  /**
+   * Combines all the root nodes for a given group into a new root node.
+   * @param {string} group
+   */
+  createRootNodeForGroup(group) {
+    const rootNodes = this.rootNodesByGroup[group];
+
+    const children = rootNodes.map(rootNode => {
+      // TODO: keep?
+      // Wrap with the name of the rootNode. Only for bundles.
+      if (group === 'scripts' && rootNode.node.children) {
+        return {
+          name: rootNode.name,
+          children: [rootNode.node],
+          resourceBytes: rootNode.node.resourceBytes,
+          unusedBytes: rootNode.node.unusedBytes,
+        };
+      }
+
+      return rootNode.node;
+    });
+
+    return {
+      name: this.documentUrl,
+      resourceBytes: children.reduce((acc, cur) => cur.resourceBytes + acc, 0),
+      unusedBytes: children.reduce((acc, cur) => (cur.unusedBytes || 0) + acc, 0),
+      children,
+    };
+  }
+
+  show() {
+    const group = 'scripts';
+    this.currentRootNode = this.createRootNodeForGroup(group);
+    const rootNodes = this.rootNodesByGroup[group];
+    renderViewModeOptions(rootNodes);
+
+    TreemapUtil.dfs(this.currentRootNode, node => {
+      // @ts-ignore: webtreemap will store `dom` on the data to speed up operations.
+      // However, when we change the underlying data representation, we need to delete
+      // all the cached DOM elements. Otherwise, the rendering will be incorrect when,
+      // for example, switching between "All JavaScript" and a specific bundle.
+      delete node.dom;
+
+      // @ts-ignore: webtreemap uses `size` to partition the treemap.
+      node.size = node.resourceBytes || 0;
+    });
+    webtreemap.sort(this.currentRootNode);
+
+    this.el.innerHTML = '';
+    this.render();
+  }
+
+  render() {
+    webtreemap.render(this.el, this.currentRootNode, {
+      padding: [18, 3, 3, 3],
+      spacing: 10,
+      caption: node => this.makeCaption(node),
+      // showChildren: node => node.children && node.children.some(c => c.resourceBytes > 1000 * 100),
```

Comment:
aye. last i remember the defaults for showChildren/showNode are surprisingly great.

Reply:
delete these lines?